### ISSUE TITLE
Installing over wireless?

### ISSUE THREAD

**Issue by nodiscc** [debian-pi/raspbian-ua-netinst#44]

**Comment by JedMeister** [comment]
This is a very cool thing. IMO you should also document which chips it supports OOTB and how to include other ones...

**Comment by goranche** [comment]
> What wlan drivers should be included in the installation? So far I have only included realtek because that's the one I use.

I was about to say "all the [working ones](http://elinux.org/RPi_USB_Wi-Fi_Adapters)", but damn, that's a large list... :smile: 

I'd say PR it, and maybe include some info on how to add additional drivers, and let others add the drivers they need... :+1:

**Comment by diederikdehaas** [comment]
> > What wlan drivers should be included in the installation? So far I have only included realtek because that's the one I use.

> I was about to say "all the working ones", but damn, that's a large list...

And then it's still incomplete :wink:  as I have an [Edimax EW-7822UAC (AC1200)](http://www.edimax.com/edimax/merchandise/merchandise_detail/data/edimax/global/wireless_adapters_ac1200_dual-band/ew-7822uac) with a [working driver](https://github.com/diederikdehaas/rtl8812AU) :stuck_out_tongue_winking_eye: 

Don't worry about the drivers for now ... that'll come. There actually [already is an issue for that](https://github.com/debian-pi/raspbian-ua-netinst/issues/52)

**Comment by goranche** [comment]
> And then it's still incomplete

that's why I say "PR it" and let the people with additional hardware do the boring work of adding drivers :angel: :wink:

**Comment by JedMeister** [comment]
> let the people with additional hardware do the boring work of adding drivers

:+1:

**Comment by diederikdehaas** [comment]
> let the people with additional hardware do the boring work of adding drivers

We could add the in-tree kernel wireless drivers to the installation image, which would add 1.4MB (tar.gz compressed) to the installation image. Motivation for this is that they would (already) be available in the installed system.
For out-of-tree kernel drivers I agree with you to let others do the work themselves. 
(Best way would be to create `\<driver\>-dkms` debian packages so wireless won't stop working when a new kernel gets installed)

One possible solution in which we could facilitate custom drivers is by letting the user create a drivers.tar.gz archive which matches the tree structure for kernel modules (ie `/lib/modules/<kernel-version>/kernel/drivers/`)

```sh
if [ -f /bootfs/config/drivers.tar.gz ] ; then
   echo -n "Loading drivers... "
   tar -zxf /bootfs/config/drivers.tar.gz -C /
   echo "OK"
   echo -n "Running depmod to update the list of modules... "
   /sbin/depmod -a
   echo "OK"
fi
```
:thought_balloon:

**Comment by Mausy5043** [comment]
I've got the installer working on WiFi.
Some manual intervention was needed though.

First of all it turns out that my WiFi-dongle needs the module `rtl8192cu` to be loaded. So, I added the parameter `drivers_to_load=rtl8192cu` in `installer-config.txt`.
But that didn't work because in the installer the module in `modules.dep` is listed as `8192cu`.
Changing to `drivers_to_load=8192cu` in `installer-config.txt` got the correct module loaded.

Then, the static command
```
wpa_supplicant -B -Dnl80211,wext -c/bootfs/config/wpa_supplicant.conf -i$ifname
```
didn't appear to work since the dongle isn't an nl80211 (and that module wasn't loaded). Changing the switch `-Dnl80211,wext` to  `-Drtl8192cu,wext` did the trick.

**Comment by Mausy5043** [comment]
I'm now stuck at this line:
    https://github.com/debian-pi/raspbian-ua-netinst/blob/v1.1.x/scripts/etc/init.d/rcS#L1032
```
chroot /rootfs /usr/bin/apt-get -y install wpasupplicant
```
produces this output:

> ```
> Reading package lists...
> Building dependency tree...
> Reading state information...
> Package wpasupplicant is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> 
> E: Package 'wpasupplicant' has no installation candidate
> FAILED !
> ```

**Comment by diederikdehaas** [comment]
> didn't appear to work since the dongle isn't an nl80211 (and that module wasn't loaded). Changing the switch -Dnl80211,wext to -Drtl8192cu,wext did the trick.

That surprises me a LOT ... and I think it's actually wrong.
According to `wpa_supplicant`'s manpage, you can get a list of valid values by doing `wpa_supplicant -h` and on my Debian Sid system that returns:
```
drivers:
  nl80211 = Linux nl80211/cfg80211
  wext = Linux wireless extensions (generic)
  wired = Wired Ethernet driver
  none = no driver (RADIUS server/WPS ER)
```
There is a reasonably chance that your (actual) driver doesn't support the nl80211 specification/interface even though it has been out for quite a while now. But then it should support `wext` as that was the previous specification/interface. 

I have no clue as to why `wpasupplicant` can't be found ...

**Comment by Mausy5043** [comment]
My guess is that `wpasupplicant` is not found because we're trying to install it using the installed system and that hasn't been `apt-get update`d yet.
I think it would be better to add `wpasupplicant` to the packages earlier in the script using the same conditionals.

**Comment by Mausy5043** [comment]
@diederikdehaas : good catch
> That surprises me a LOT ... and I think it's actually wrong.
According to `wpa_supplicant`'s manpage, you can get a list of valid values by doing `wpa_supplicant -h` and on my Debian Sid system that returns:
```
drivers:
  nl80211 = Linux nl80211/cfg80211
  wext = Linux wireless extensions (generic)
  wired = Wired Ethernet driver
  none = no driver (RADIUS server/WPS ER)
```

In fact anything other than one of those four drivers gets silently ignored. Because there's also `wext` there it will pick that. So, I'm concluding from that that `-Dnl80211,wext` is not going to work (always). If the `nl80211` is not supported `wpa_supplicant` won't try `wext` on its own. So we have to force that in the script.

**Comment by Mausy5043** [comment]
I'm starting to wonder if we **have** to specify the `-D` switch. manpage says it's optional.

**Comment by Mausy5043** [comment]
I am led to believe that this issue was solved by the merging of PR #397 into `v1.1.x`. ;-)
As a result of that belief, I am now closing this issue.
If you feel the closure was in error, please, feel free to re-open the issue.
